Install 3rd-pary softwares from binary release.

hongquan opened the pull request:
Install from binary release instead of building from source, to save disk space.

All the Python libraries which are C-based, are built manually and uploaded to our repo https://repo.fury.io/fossasia/ by us. It won't be built on RPi, so that we don't have to install compiler, library header files to RPi.

The repo only contains prebuilt Python packages for Python3.5-armv7l (for RPi) and Python3.5-x86_64  (for TravisCI). So this script will fail on Ubuntu, which may have different Python (3.6). The developer working on Ubuntu should be able to install the dependencies himself.

cweitat commented:
Travis again

hongquan commented:
Yes, trying to fix Travis. The Travis is running part of old installation script, which tries to install dependencies software by building from source. Now, I need to build wheel packages for the environment that Travis is running on (Python 3.5, x86_64).


With this PR, I can manage to shrink the SD image file to 1.3GB (compressed). Following steps also need to be done:
- Fill zero to unused space, with `zerofree` tool.
- When using `dd` to copy disk to img file, add `conv=sparse` to take advantage of the zero-filled space.
- Compress the img file with `pixz` tool.

hongquan commented:
It seems that the Access Point feature doesn't work after installation and reboot.

stealthanthrax commented:
@hongquan , what issue are you getting?? It works on my side.
> It seems that the Access Point feature doesn't work after installation and reboot.

stealthanthrax commented on requirements-hw.txt:
```diff
@@ -1,11 +1,11 @@
-speechRecognition==3.6.5
-pocketsphinx==0.1.3
+snowboy
```
@hongquan , has your PR been merged in the snowboy repo??

hongquan replied:
@stealthanthrax Not yet, but we tell `pip` to install `snowboy` from the wheel package we uploaded to https://repo.fury.io/fossasia/.

stealthanthrax commented on install.sh:
```diff
@@ -55,68 +36,28 @@ function install_seed_voicecard_driver()
-        git clone https://github.com/fossasia/susi_server.git
-        git clone https://github.com/fossasia/susi_skill_data.git
+        git clone --recurse-submodules https://github.com/fossasia/susi_server.git
+        git clone --recurse-submodules https://github.com/fossasia/susi_skill_data.git
```
@hongquan , we do not need to use `--recursive-submodules` for susi_skill_data repo

hongquan commented:
> what issue are you getting?? It works on my side.

@stealthanthrax Haven't tested that. Just see that, after RPI is reboot, the green LED light is not ON.
I'm noting here so that I will check it later.

hongquan commented:
TravisCI failed due to misconfiguration on TravisCI side: They added MongoDB repo, which is refused by APT due to weak key.

hongquan commented on install.sh:
```diff
@@ -55,68 +36,30 @@ function install_seed_voicecard_driver()
-        git clone https://github.com/fossasia/susi_server.git
+        git clone --recurse-submodules https://github.com/fossasia/susi_server.git
         git clone https://github.com/fossasia/susi_skill_data.git
+        # The .git folder is big. Delete it (we don't do susi_server deveplopment here, so no need to keep it)
```
@stealthanthrax We will get update by different method. I expect *susi_server* is released by version, and we will download it by version (not continuous update).

I also expect `susi_server` to be built and released in form of Debian package, so that we don't have to install `gradle` tool, other Java libraries just to build it on RPi.

For now, when *susi_server* is not provided in the form I want, we will include a fixed version of *susi_server* in each version of SD card image. In case the developer wants to update it without reflashing RPi with new SD image, he can just download tarball of *susi_server*.

stealthanthrax replied:
Okay @hongquan , makes sense 👍

hongquan commented:
I've just added a note in description for Ubuntu user.

Maybe we have a separate install script for Ubuntu user?

hongquan commented:
Found some error with compressing source folder for "factory reset". Fixing it...

hongquan commented:
This is the last messages when running _access_point/wap.sh_ at the end of install script.

```
Setting up hostapd (2:2.6-14) ...
Created symlink /etc/systemd/system/hostapd.service → /dev/null.
Setting up dnsmasq (2.76-5+rpt1+deb9u1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/dnsmasq.service → /lib/systemd/system/dnsmasq.service.
Processing triggers for systemd (232-25+deb9u4) ...
Synchronizing state of hostapd.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable hostapd
Failed to enable unit: Unit file /etc/systemd/system/hostapd.service is masked.
Synchronizing state of dnsmasq.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable dnsmasq
Failed to start hostapd.service: Unit hostapd.service is masked.
```

As you can see, APT warned that _hostapd.service_ cannot be started, I think it is related to RPi failure after reboot.

hongquan commented:
After rebooting, I don't see RPi's access point, also don't see it in LAN.

Plugging its SD card to laptop, I found this log: https://paste.ee/p/WPg6x

hongquan commented:
The cause of breakage of _hostapd_ and RPI failure is due to the packages in ReSpeaker repo (https://github.com/respeaker/deb), which I added to install script. Removed it and RPi is fixed.

hongquan commented:
The new image can fit in 4GB SD card.

![image](https://user-images.githubusercontent.com/314607/44484012-c702c780-a676-11e8-9337-1a155d0bbdd1.png)

hongquan commented:
First working image here: http://quan.hoabinh.vn/Downloads/susi-linux-sd/susi_linux_20180823.img.xz

Note: It contains some my personal settings and software.

hongquan commented:
Other link (Singaporean server): https://agriconnect.vn/media/susi-linux-sd/susi_linux_20180823.img.xz
